packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -1679,17 +1679,16 @@ export class Task {
     }
 
     private updateDefinitionBasedOnExecution(): void {
-        this.taskDefinition = undefined;
         if (this.taskExecution instanceof ProcessExecution) {
-            this.taskDefinition = {
+            this.taskDefinition = Object.assign(this.taskDefinition, {
```

Comment:
Using `Object.assign` I don't think you need the assignation before, as the fields will already be injected into the object.

Reply:
are you suggesting to do the following ?
```ts
this.taskDefinition.type = 'shell';
this.taskDefinition.id = .... ;
```

Reply:
resolved